examples/nft-d/nft.go

```diff
@@ -425,3 +425,61 @@ func Update(nef, manifest []byte) {
 	}
 	management.Update(nef, manifest)
 }
+
+// getOwnerOf returns the current owner of the specified token or panics if token
+// ID is invalid. The owner is stored as a value of the token key (prefix + token ID).
+func getOwnerOf(ctx storage.Context, token []byte) interop.Hash160 {
+	key := mkTokenKey(token)
+	val := storage.Get(ctx, key)
+	if val == nil {
+		panic("no token found")
+	}
+	return val.(interop.Hash160)
+}
+
+// RoyaltyRecipient contains information about the recipient and the royalty amount.
+type RoyaltyRecipient struct {
+	Address interop.Hash160
+	Amount  int
+}
+
+// RoyaltyInfo returns a list of royalty recipients and the corresponding royalty amounts.
+func RoyaltyInfo(tokenID []byte, royaltyToken interop.Hash160, salePrice int) []RoyaltyRecipient {
+	if salePrice <= 0 {
+		panic("sale price must be positive")
+	}
+
+	callingHash := runtime.GetExecutingScriptHash()
+	if !royaltyToken.Equals(callingHash) {
+		panic("invalid royalty token")
+	}
+
+	ctx := storage.GetReadOnlyContext()
+	owner := getOwnerOf(ctx, tokenID)
+	if owner == nil {
+		panic("invalid token ID")
+	}
+
+	royaltyInfoKey := append([]byte("royaltyInfo:"), tokenID...)
+	val := storage.Get(ctx, royaltyInfoKey)
+
+	var recipients []RoyaltyRecipient
+	if val == nil {
+		recipients = []RoyaltyRecipient{
+			{
+				Address: owner,
```

Comment:
Added 
```
{
				Address: contractOwner,
				Amount:  (salePrice * 5) / 100,
			},
			
```

multiple owners==	multiple recipients?

Reply:
> Address: contractOwner,

It's not correct because contract owner is not the token owner, and here we're talking about *royalty* payment, i.e. payment to token owner.

> multiple owners== multiple recipients?

It depends on the contract logic, you may implement as you wish. But this comment is related to `Address: owner,`, because the previous code that retrieved `owner` was buggy.